pkg/reconciler/build/resources/pod.go

```diff
@@ -491,3 +525,397 @@ func getFailureMessage(pod *corev1.Pod) string {
 	// Lastly fall back on a generic error message.
 	return "build failed for unspecified reasons."
 }
+
+// from build-pipeline
```

Comment:
can this be broken up into multiple files? also i think this comment needs more details